geoapps/grid_creation/application.py

```diff
@@ -0,0 +1,383 @@
+        )(self.update_params)
+        self.app.callback(
+            Output(component_id="live_link", component_property="value"),
+            Input(component_id="export", component_property="n_clicks"),
+            prevent_initial_call=True,
+        )(self.create_block_model)
+
+    def update_params(
+        self,
+        filename,
+        contents,
+        new_grid,
+        objects,
+        xy_reference,
+        cell_size_x,
+        cell_size_y,
+        cell_size_z,
+        depth_core,
+        horizontal_padding,
+        bottom_padding,
+        expansion_fact,
+        live_link,
+        monitoring_directory,
+    ):
+        param_list = [
+            "new_grid",
+            "objects_name",
+            "objects_options",
+            "xy_reference_name",
+            "xy_reference_options",
+            "call_size_x",
+            "call_size_y",
+            "call_size_z",
+            "depth_core",
+            "horizontal_padding",
+            "bottom_padding",
+            "expansion_fact",
+            "monitoring_directory",
+            "filename",
+            "contents",
+        ]
+        cell_size_x, cell_size_y, cell_size_z = (
+            float(cell_size_x),
+            float(cell_size_y),
+            float(cell_size_z),
+        )
+        horizontal_padding, bottom_padding = float(horizontal_padding), float(
+            bottom_padding
+        )
+        depth_core, expansion_fact = float(depth_core), float(expansion_fact)
+
+        trigger = callback_context.triggered[0]["prop_id"].split(".")[0]
+        update_dict = {}
+        if trigger == "upload":
+            if (filename.endswith("ui.json") or filename.endswith(".geoh5")) and (
+                contents is not None
+            ):
+                if filename.endswith(".ui.json"):
+                    update_dict.update(self.update_from_ui_json(contents, param_list))
+                    ws = update_dict["geoh5"]
+                elif filename.endswith(".geoh5"):
+                    content_type, content_string = contents.split(",")
+                    decoded = io.BytesIO(base64.b64decode(content_string))
+                    ws = Workspace(decoded)
+                update_dict.update(self.update_object_options(ws, "objects"))
+                update_dict.update(self.update_object_options(ws, "xy_reference"))
+                update_dict["filename"] = None
+                update_dict["contents"] = None
+            else:
+                print("Uploaded file must be a workspace or ui.json.")
+        elif trigger + "_name" in param_list:
+            update_dict[trigger + "_name"] = locals()[trigger]
+        elif trigger != "":
+            update_dict[trigger] = locals()[trigger]
+
+        self.update_param_dict(update_dict)
+        outputs = self.get_outputs(param_list, update_dict)
+
+        return outputs
+
+    def create_block_model(self, n_clicks):
```

Comment:
Doesn't look like n_clicks is used in the function body.  Also the rest of the applications would call this method trigger_click.  I'm tempted to try to keep it consistent with the rest of the apps, but this is a more descriptive name so maybe better to update the other apps?  @domfournier  what do you think?

Reply:
Can you just pass `_` if you need a placeholder for the callback.